### PR TITLE
Proposal for type validation test runner

### PR DESCRIPTION
**Pull request by kwonoj**
relates to #661 

This PR include small code snippet to check basic type definitions as discussed in issue.
There'll be chances of improvement to have additional type detection for verbose output, would like to consider this as starting point.

No actual tests & CI integration included yet.

**Output example**
image

### CODE REVIEW

**Comment by kwonoj** [comment]
cc @robwormald , would like to ask suggestions for output format or any other.

File: types/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      var lines = file.split('\n');
+      _.forEach(value, function (diagnostic) {
+        var pos = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
+        var message = ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n');
+
+        console.log('Error : '.red + message);
+        console.log(lines[pos.line].gray);
+        console.log(_.repeat(' ', pos.character) + '^');
+        console.log();
```
Comment (staltz) [inline types/index.js]: How about `console.error()`?

Comment (kwonoj) [inline types/index.js]: Yes, that could be way. Will update.

**Comment by kwonoj** [comment]
Small update suggestion from @staltz to use console.error with minor output changes.

image

**Comment by blesh** [comment]
This is interesting. How long does it take to run, on average? (I'll pull it down sometime soon and give it a try, too)

**Comment by kwonoj** [comment]
What I did was just small code snippet does not reflect real world cases (having lot of files, or codes) though, it took about a second. For comparison reference, on same system typescript build takes around 14 sec, unit test takes around 2.3 second.

**Comment by blesh** [comment]
@trxcllnt ... what are your thoughts?

**Comment by kwonoj** [comment]
Marked this as blocked while exploring feasibility of migrating unit test cases to typescript based.

**Comment by blesh** [comment]
@kwonoj I'm going to close this for now since discussion here is dead. We can open a new issue for it to create more discussion, or we can discuss off-line and open an issue when we have some plans outlined.

**Comment by kwonoj** [comment]
Agreed, thanks.

**Pull request closed without merging**